src/client.rs

```diff
@@ -186,87 +181,91 @@ pub struct CommandExecutedResult {
     pub output: String,
     /// The unix exit status (`$?` in bash).
     pub exit_status: u32,
-    /// The actual address of the [`ToSocketAddrs`] argument that was used for the ssh connection.
-    pub actual_address: SocketAddr,
+}
+
+impl CommandExecutedResult {
+    fn new(output: String, exit_status: u32) -> Self {
+        Self {
+            output,
+            exit_status,
+        }
+    }
 }
 
 #[derive(Clone)]
 struct ClientHandler;
 
+impl ClientHandler {
+    fn new() -> Self {
+        Self {}
+    }
+}
```

Comment:
Unrelated to the connection code, I removed this since you could just use the struct constructor. I find this to be easier to read and it is not exposed to in the api, thus it doesn't matter if we remove it. I suggest removing it again.